File: server/src/engine/factories/playerfactory.ts

import Transform = require("../components/transform");
import Player = require("../components/player");
import Render = require("../components/render");
import EntityManager = require("../entitymanager");

class PlayerFactory {
    static new(em: EntityManager, id: number, position: Vector2): number {
        var player = em.createEntity("Player " + id);
        em.addComponent(player, new Transform(position, 75));
        em.addComponent(player, new Player("Player " + id.toString()));
        em.addComponent(player, new Render("Player"));
        return player;
    }
}

export = PlayerFactory

File: sharpServer/Engine/Factories/PlayerFactory.cs
namespace SharpServer.Engine.Factories
{
    public static class PlayerFactory
    {
        public static uint Create(int id, Vector2 position)
        {
            var player = EntityBaseFactory.Create(position, "Player " + id);
            EntityManager.AddComponents(player, 
                new Player("Player " + id),
                new Render("Player", true, "Shitty hero", "Hero", "Really shitty hero"),
                new Moving(100),
                new Shape(75),
                new Health(100));
            return player;
        }
    }
}